app/components/navigation_header/index.tsx

```diff
@@ -58,58 +61,73 @@ const NavigationHeader = ({
     const insets = useSafeAreaInsets();
     const styles = getStyleSheet(theme);
 
-    const {largeHeight, defaultHeight} = useHeaderHeight();
+    const {largeHeight, defaultHeight} = useHeaderHeight(hasSearch);
     const containerHeight = useAnimatedStyle(() => {
         const minHeight = defaultHeight + insets.top;
         const value = -(scrollValue?.value || 0);
-        const height = ((isLargeTitle ? largeHeight : defaultHeight)) + value + insets.top;
+        const calculatedHeight = (isLargeTitle ? largeHeight : defaultHeight) + value;
+        const height = lockValue?.value ? lockValue.value : calculatedHeight;
         return {
             height: Math.max(height, minHeight),
             minHeight,
             maxHeight: largeHeight + insets.top + MAX_OVERSCROLL,
         };
     });
 
+    const minScrollValue = useDerivedValue(() => scrollValue?.value || 0, [scrollValue]);
+    const heightOffset = useDerivedValue(() => largeHeight - defaultHeight, [largeHeight, defaultHeight]);
+
+    const translateY = useDerivedValue(() => (
```

Comment:
I extracted these two calculations from their components and pass them in as props. I think the is easier to see how the new lockValue value is used for each compononent.